### PR TITLE
refactor: cleanup common code and tests

### PR DESCRIPTION
**Pull request by vlavoie**
## Description
This refactoring was separated from the work done on issue #4. It cleans up and polishes up a lot of the code.
Additionally it improves the unit test behavior to actually use the REST channel through a mock client and refactors the code to allow passing a different client to the application.

## Related issues
None, extracted from changes meant for #4 

## Checklist

- [x] PR title follows [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) guidelines 
- [x] All relevant PR sections are populated, irrelevant ones are removed

### CODE REVIEW

File: rasa_integration_testing/application.py [hunk 1]
```diff
@@ -2,78 +2,73 @@
...
+        default=DEFAULT_ASYNC_CHUNK_SIZE,
+        help="Size of asynchronous test chunks.",
     )
+    @click.option("-o", "--output", type=click.File("w"), default=sys.stdout)
+    @click.argument("scenarios_glob", required=False, default=SCENARIOS_GLOB)
+    def cli(
+        tests_path: str, chunk_size: int, output: TextIO, scenarios_glob: str
+    ) -> int:
+        folder_path = Path(tests_path)
+        configuration = Configuration(folder_path.joinpath(TEST_CONFIG_FILE))
```
Comment (fparga) [inline rasa_integration_testing/application.py:48]: Is there an advantage to use `joinpath` instead the `/` operator?

Such as:
```python
         configuration = Configuration(folder_path / TEST_CONFIG_FILE)
```

**Comment by fparga** [comment]
If I understand correctly this is a preamble to another PR that will address #4 ?

File: rasa_integration_testing/application.py (hunk 1 above)
Comment (vlavoie) [reply rasa_integration_testing/application.py:48]: Does pathlib override the `/` operator to accomplish the same behavior? I actually didn't know it supported this syntax

**Comment by vlavoie** [comment]
> 
> 
> If I understand correctly this is a preamble to another PR that will address #4?
Yes, it will include more changes to the runner but I didn't want to overly clog up those changes with renames and small refactors.

File: rasa_integration_testing/application.py (hunk 1 above)
Comment (vlavoie) [reply rasa_integration_testing/application.py:48]: FYI, after looking at the sources, they are indeed identical. 
image

**Pull request merged**